capstone-rs/src/instruction.rs

```diff
@@ -114,6 +114,14 @@ impl<'a> Instructions<'a> {
     }
 }
 
+impl<'a> core::ops::Deref for Instructions<'a> {
+    type Target = [Insn<'a>];
+
+    fn deref(&self) -> &[Insn<'a>] {
+        unsafe { core::slice::from_raw_parts(self.0.as_ptr() as *const Insn, self.0.len()) }
```

Comment:
Could you add a comment here explaining why this is safe (it doesn't have to be long).
I'm thinking something like:

```rust
// CLAIM: `Insn` has the same representation as `cs_insn` because of the `repr(transparent)` over `Insn`
```
```

Reply:
Didn't get a chance to see this before rebasing abd pushing my changes. I changed my implementation so it's closer to what the standard library does for similar things. Here's a similar cast in the `std::cell::Cell` implementation:

```rust
#[repr(transparent)]
pub struct Cell<T: ?Sized> {
    value: UnsafeCell<T>, // NOTE: UnsafeCell is also repr(transparent)
}

impl<T> Cell<[T]> {
    pub fn as_slice_of_cells(&self) -> &[Cell<T>] {
        // SAFETY: `Cell<T>` has the same memory layout as `T`.
        unsafe { &*(self as *const Cell<[T]> as *const [Cell<T>]) }
    }
}
```